dev/devicelab/lib/tasks/plugin_tests.dart

```diff
@@ -151,6 +156,79 @@ class $dartPluginClass {
     }
   }
 
+  /// Converts an iOS/macOS plugin created from the standard template to a shared
+  /// darwin directory plugin.
+  Future<void> convertDefaultPluginToSharedDarwinPlugin() async {
+    // Convert the metadata.
+    final File pubspec = pubspecFile;
+    String pubspecContent = await pubspec.readAsString();
+    const String originalIOSKey = '\n      ios:\n';
+    const String originalMacOSKey = '\n      macos:\n';
+    if (!pubspecContent.contains(originalIOSKey) || !pubspecContent.contains(originalMacOSKey)) {
+      print(pubspecContent);
+      throw TaskResult.failure('Missing expected darwin platform plugin keys');
+    }
+    pubspecContent = pubspecContent.replaceAll(
+        originalIOSKey,
+        '$originalIOSKey        sharedDarwinSource: true\n'
+    );
+    pubspecContent = pubspecContent.replaceAll(
+        originalMacOSKey,
+        '$originalMacOSKey        sharedDarwinSource: true\n'
+    );
+    await pubspec.writeAsString(pubspecContent, flush: true);
+
+    // Copy ios to darwin, and delete macos.
+    final Directory iosDir = Directory(path.join(rootPath, 'ios'));
+    final Directory darwinDir = Directory(path.join(rootPath, 'darwin'));
+    recursiveCopy(iosDir, darwinDir);
+
+    await iosDir.delete(recursive: true);
+    await Directory(path.join(rootPath, 'macos')).delete(recursive: true);
+
+    final File podspec = File(path.join(darwinDir.path, '$name.podspec'));
+    String podspecContent = await podspec.readAsString();
+    if (!podspecContent.contains('s.platform =')) {
+      print(podspecContent);
+      throw TaskResult.failure('Missing expected podspec platform');
+    }
+
+    // Remove "s.platform = :ios" to work on all platforms, including macOS.
+    podspecContent = podspecContent.replaceFirst(RegExp(r'.*s\.platform.*'), '');
+    podspecContent = podspecContent.replaceFirst("s.dependency 'Flutter'", "s.ios.dependency 'Flutter'\ns.osx.dependency 'FlutterMacOS'");
+
+    await podspec.writeAsString(podspecContent, flush: true);
+
+    // Make PlugintestPlugin.swift compile on iOS and macOS with target conditionals.
+    final String pluginClass = '${name[0].toUpperCase()}${name.substring(1)}Plugin';
+    print('pluginClass: $pluginClass');
+    final File pluginRegister = File(path.join(darwinDir.path, 'Classes', '$pluginClass.swift'));
+    final String pluginRegisterContent = '''
+#if os(macOS)
+import FlutterMacOS
+#elseif os(iOS)
+import Flutter
+#endif
+
+public class $pluginClass: NSObject, FlutterPlugin {
+  public static func register(with registrar: FlutterPluginRegistrar) {
+#if os(macOS)
+    let channel = FlutterMethodChannel(name: "$name", binaryMessenger: registrar.messenger)
+#elseif os(iOS)
+    let channel = FlutterMethodChannel(name: "$name", binaryMessenger: registrar.messenger())
```

Comment:
I really want to see if we can fix this; it bit me recently too. For Obj-C the change is transparent, but it's unclear to me if there's a way to change it without breaking Swift plugins. (Not for this PR, just venting.)